Product/Production/Services/PatientDiscoveryCore/src/main/java/gov/hhs/fha/nhinc/patientdiscovery/audit/transform/AbstractPatientDiscoveryAuditTransforms.java

```diff
@@ -0,0 +1,220 @@
+    }
+
+    @Override
+    protected String getServiceEventActionCodeRequestor() {
+        return PatientDiscoveryAuditTransformsConstants.EVENT_ACTION_CODE_REQUESTOR;
+    }
+
+    @Override
+    protected String getServiceEventActionCodeResponder() {
+        return PatientDiscoveryAuditTransformsConstants.EVENT_ACTION_CODE_RESPONDER;
+    }
+
+    protected AuditMessageType getPatientParticipantObjectIdentificationForResponse(PRPAIN201306UV02 response,
+        AuditMessageType auditMsg) {
+
+        List<II> oII = PRPAIN201306UV02Parser.getPatientIds(response);
+        if (oII != null && oII.size() > 0) {
+            for (II entry : oII) {
+                if (entry == null) {
+                    createPatientParticipantObjectIdentification(auditMsg, null, null);
+                } else {
+                    createPatientParticipantObjectIdentification(auditMsg, isNotNullish(entry.getRoot()) ? entry.
+                        getRoot().trim() : null, isNotNullish(entry.getRoot()) ? entry.getExtension().trim() : null);
+                }
+            }
+        } else {
+            createPatientParticipantObjectIdentification(auditMsg, null, null);
+        }
+
+        return auditMsg;
+    }
+
+    protected AuditMessageType getQueryParamsParticipantObjectIdentificationForResponse(PRPAIN201306UV02 response,
+        AuditMessageType auditMsg) throws JAXBException {
+
+        ParticipantObjectIdentificationType participantObject = buildBaseParticipantObjectIdentificationType(
+            PRPAIN201306UV02Parser.getQueryId(response));
+        participantObject.setParticipantObjectQuery(getParticipantObjectQueryForResponse(response));
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+        return auditMsg;
+    }
+
+    protected AuditMessageType getPatientParticipantObjectIdentificationForRequest(PRPAIN201305UV02 request,
+        AuditMessageType auditMsg) {
+
+        // Set the Participant Object Id (patient id)
+        II oII = PRPAIN201305UV02Parser.getPatientId(request);
+        if (oII != null && oII.getRoot() != null && oII.getExtension() != null && !oII.getRoot().isEmpty()
+            && !oII.getExtension().isEmpty()) {
+
+            createPatientParticipantObjectIdentification(auditMsg, oII.getRoot(), oII.getExtension());
+        } else {
+            createPatientParticipantObjectIdentification(auditMsg, null, null);
+        }
+
+        return auditMsg;
+    }
+
+    private byte[] getParticipantObjectQueryForResponse(PRPAIN201306UV02 response) throws JAXBException {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        if (response != null && response.getControlActProcess() != null
+            && response.getControlActProcess().getQueryByParameter() != null) {
+            getMarshaller().marshal(response.getControlActProcess().getQueryByParameter(), baos);
+        }
+        return baos.toByteArray();
+    }
+
+    private boolean isNotNullish(String value) {
```

Comment:
Needed a trim check on value as well. Currently NullChecker does not trim the string in either isNullish or isNotNullish. 
Will add a isNullishIgnoreSpace in NullChecker class and use it in AbstractPatientDiscoveryAuditTransforms